numpy/_core/tests/test_stringdtype.py

```diff
@@ -828,6 +828,31 @@ def test_add_promoter(string_list):
         assert_array_equal(op + arr, lresult)
         assert_array_equal(arr + op, rresult)
 
+    # The promoter should be able to handle things if users pass `dtype=`
+    res = np.add("hello", string_list, dtype=StringDType)
```

Comment:
Probably not worth using the `dtype` fixture for this since `na_object` and `coerce` doesn't matter, but maybe worth making `dtype` a parameter of the test that can either by `StringDType` or `StringDType()`. I could also see perhaps defining a `class_or_instance` fixture and using that in a few other places in this file where we just test with `StringDType()` or `"T"`.

Reply:
That doesn't work.  Signatures are DType classes.  It should work at least for `"T"`, but otherwise need to look into logic to say "this is OK".
I am not sure it should be, but it is a different issue in either case.

Reply:
Yeah, not hat happens much earlier, I explicitly allowed the singleton instancs of legacy dtypes (or maybe all singleton instances, not sure), because otherwise things would be tricky.

But, we have the "give me the DType" now also, which maybe (not sure!) makes `T` work.  It _should_ make `T` work in either case, though.